lib/msf/util/exe.rb

```diff
@@ -889,9 +895,10 @@ def self.to_osx_app(exe, opts = {})
     zip.add_file("#{app_name}/Contents/", '')
     zip.add_file("#{app_name}/Contents/MacOS/", '')
     zip.add_file("#{app_name}/Contents/Resources/", '')
-    zip.add_file("#{app_name}/Contents/MacOS/#{exe_name}", exe)
+    zip.add_file("#{app_name}/Contents/MacOS/#{exe_name}", File.new(macho, 'r').read)
```

Comment:
@jmartin-r7
Please check see these changes: https://github.com/rapid7/metasploit-framework/compare/master...timwr:fix_osx_app?expand=1
https://github.com/rapid7/rex-zip/compare/master...timwr:expose_file_attrs?expand=1
Unfortunately the file is still fat (not Unix) so it only works because it thinks the file is a directory :/
I'll fix it properly tmw